main.py

```diff
@@ -34,14 +35,14 @@ def main(on_cap, check_network_close):
             time.sleep(RATE)
             continue
 
-        frame_end_ts = ts + RATE
+        frame_end_ts = (ts or read_ts) + RATE
```

Comment:
> (ts or read_ts)

This is fine for now but i'd rather that we pass in speed into stream.py instead. In the future i would like to have source_fps and ocr_fps separated; where both can have automatic options:

* Capture cards present FPS, so with AUTO we can set it to capture_card.fps * [2 if (interlace != discard)]
* OCR_FPS should be 30 or 60 by default, with using AUTO to first check what the capture method's FPS is, followed by dynamically scaling fps from lagtime (e.g. peasant computer can OCR at 10fps to get a smoother output)

So for after this PR, either:
* add `capture_fps` into config, rename `performance.scan_rate` to `ocr_fps` into config, then make stream.py respect it (older classes can ignore it, that's a future problem) and main.py use ocr_fps
* more hacky: just make stream.py read the config and use `performance.scan_rate`


But for now lets just leave it unless you love doing more work :D

Reply:
Noted! Let's merge this for now since it works, and we can improve in future PR. 

Oh btw, using different `source_fps` and `ocr_fps` might require extra changes to stream.py 🤔 . Otherwise, at the beginning of the game, until the frame buffer fills up, ocr would read continuous frames but at a "slow motion rate". It should stabilize once the frame buffer is full (dequeue will drop the frames), but the beginning of capture might look weird.

Reply:
Yep lets not worry about it for now since its *not a trivial 5 line change*.

I would assume stream.py would simply return blank or the first frame for the first [buffer_size] frames and then after buffer is say 10~ long it would behave normally. But lets leave that as a future problem.

Reply:
Hmm, If we wait for buffer to be filled, then the delay should be related to the buffer size, not the ratio of read and scan rates 🤔  Delay of stream would be be (buffer_size / source_fps). With the current (completely arbitrary)  buffer size of 150, a 30fps stream would be delayed by 5s before we start ocr/rendering.

Reply:
` If we wait for buffer to be filled`
The `fill before allowing frame ejection` could be say 20-30 and the full buffer could be 150. then if the buffer gets completely empty we reset it to not remove items from buffer till it hits 20-30 again. This is similar to how streaming video in the first place works:
1) pause until buffer is at least 1s (return buffer[0], don't pop())
2) (loop) receive from internet + play at realtime rate  (buffer[0].pop(), return popped)
3) if buffer hits 0, go to step 1

This assumes an unbounded buffer. If you have correct timestamps and we implement ocr_rate correctly (its not currently correct), an ocr_rate of 10 and stream_rate of 30 would mean ocr just skips 3 frames every scan and realtime still stays "correct". It would 100% wreak havoc on any server side stuff that doesnt want skipped frames though, but if the ocr isn't fast enough because peasant computer, real-time is probably more important than having a growing stream lag.

It's getting more and more complicated, but then the server would have to support determining if the computer is "dropping frames" or not and disabling effects like field-based tetris detection, since that doesnt work if you're dropping frames like crazy.